### PR TITLE
Move dependency warnings from compiler to maven plugin

### PR DESCRIPTION
**Pull request by fwellers**
* Pass dependency warnings via a file from compiler to maven plugin
* introduce a "warningsOutputDirectory" to the compiler to specify where
  to save dependency warnings
* delete isTestCompile option in compiler
* maven plugin is now responsible for displaying error messages

### CODE REVIEW

File: jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java [hunk 1]
```diff
@@ -177,11 +182,11 @@
    * In ECMAScript, initializer values are assigned to all 'undefined' arguments.
    * In AS3, initializer values are assigned only if you call a method with less arguments.
    * An example would be
-   *     function foo(bar: string = "default"): string {
```
Comment (fwienber) [inline jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java]: If you want to reformat code, that's fine, but for review, it would help a lot if you'd do that is a separate commit!

File: jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java [hunk 2]
```diff
@@ -385,8 +396,9 @@ protected JoocConfiguration createJoocConfiguration(Log log) throws MojoExecutio
     configuration.setOutputDirectory(getClassesOutputDirectory());
     configuration.setLocalizedOutputDirectory(getLocalizedOutputDirectory());
     configuration.setApiOutputDirectory(getApiOutputDirectory());
-    configuration.setSourcesAreTests(isTestRun());
-    configuration.setFindUnusedDependencies(findUnusedDependencies());
+    configuration.setFindUnusedDependencies(findUnusedDependencies(staleMillis));
+    //todo: use proper directory
+    configuration.setWarningsOutputDirectory("dependencyWarnings");
```
Comment (fwienber) [inline jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java]: I thought the options were
- either a warnings output directory with multiple text files with well-known names
- or a single JSON file for the report
Now you have a directory with one file with a well-known name.
The configuration should be the File, not the directory.

File: jangaroo-tools-api/jangaroo-compiler-api/src/main/java/net/jangaroo/jooc/config/JoocOptions.java [hunk 3]
```diff
@@ -38,7 +38,7 @@
 
   boolean isFindUnusedDependencies();
 
-  boolean isSourcesAreTests();
+  String getWarningsOutputDirectory();
```
Comment (fwienber) [inline jangaroo-tools-api/jangaroo-compiler-api/src/main/java/net/jangaroo/jooc/config/JoocOptions.java]: See above. I'd call the new option `dependencyReportOutputFile`.

**Review by fwienber — CHANGES_REQUESTED** [review]

File: jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java (hunk 1 above)
Comment (fwienber) [reply jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java]: No need to change that now, but please do so in the future.

Comment (fwellers) [reply jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java]: It was due to auto-formatting in my file. I tried to undo it, but it failed apparently.

File: jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java (hunk 2 above)
Comment (fwellers) [reply jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java]: Do you have a recommendation for a default path?

Comment (fwienber) [reply jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java]: Not really, you could put it in a `target/dependencyReports` sub-directory in order not to "pollute" the top-level `target` directory, but since it is just one file, it does not really make a difference.

File: jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java [hunk 4]
```diff
@@ -292,6 +301,8 @@ public void warning(String msg) {
 
     int result = compile(jooc);
 
+    printDependencyWarnings(configuration);
+    Paths.get(configuration.getWarningsOutputDirectory(), "dependencyWarnings").toFile().delete();
```
Comment (fwienber) [inline jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java]: What if someone (mis)configures no `warningsOutputDirectory` (or the new `dependencyReportOutputFile`), wouldn't that result in a NPE here?

Comment (fwellers) [reply jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java]: I changed it so there are no dependency checks if there is no provided path. I also added a small note about that in the JoocConfiguration to the property.

File: jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java (hunk 2 above)
Comment (fwellers) [reply jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java]: Done

File: jangaroo-tools-api/jangaroo-compiler-api/src/main/java/net/jangaroo/jooc/config/JoocOptions.java (hunk 3 above)
Comment (fwellers) [reply jangaroo-tools-api/jangaroo-compiler-api/src/main/java/net/jangaroo/jooc/config/JoocOptions.java]: done

File: jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java [hunk 5]
```diff
@@ -385,8 +400,9 @@ protected JoocConfiguration createJoocConfiguration(Log log) throws MojoExecutio
     configuration.setOutputDirectory(getClassesOutputDirectory());
     configuration.setLocalizedOutputDirectory(getLocalizedOutputDirectory());
     configuration.setApiOutputDirectory(getApiOutputDirectory());
-    configuration.setSourcesAreTests(isTestRun());
-    configuration.setFindUnusedDependencies(findUnusedDependencies());
+    configuration.setFindUnusedDependencies(findUnusedDependencies(staleMillis));
+    //todo: use proper directory
+    configuration.setDependencyReportOutputFile("target/dependencyReports/dependencyWarnings");
```
Comment (fwienber) [inline jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java]: What about some file extension like `.txt`? Could make it easier to look into the file if someone's curious.

**Review by fwienber — CHANGES_REQUESTED** [review]
Just add the file extension and you are good to go! 👍

File: jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java (hunk 5 above)
Comment (fwellers) [reply jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/AbstractCompilerMojo.java]: Coming from linux i didnt really think about any file extensions. I can add that

**Review by fwienber — APPROVED** [review]
